modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/PageRenderer/EditorVariant.es.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import React, {useContext, useRef} from 'react';
+
+import {DND_ORIGIN_TYPE, useDrop} from '../../hooks/useDrop.es';
+import {ParentFieldContext} from '../Field/ParentFieldContext.es';
+import {Actions, ActionsControls, useActions} from './Actions.es';
+import * as DefaultVariant from './DefaultVariant.es';
+import {Placeholder} from './Placeholder.es';
+
+export const Column = ({
+	activePage,
+	allowNestedFields,
+	children,
+	column,
+	editable,
+	index,
+	pageIndex,
+	rowIndex,
+}) => {
+	const parentField = useContext(ParentFieldContext);
+
+	const actionsRef = useRef(null);
+	const columnRef = useRef(null);
+
+	const {activeField, hoveredField} = useActions();
+
+	const {drop, overTarget} = useDrop({
+		columnIndex: index,
+		fieldName: column.fields[0]?.fieldName,
+		origin: DND_ORIGIN_TYPE.FIELD,
+		pageIndex,
+		parentField,
+		rowIndex,
+	});
+
+	if (editable && column.fields.length === 0 && activePage === pageIndex) {
+		return (
+			<Placeholder
+				columnIndex={index}
+				pageIndex={pageIndex}
+				rowIndex={rowIndex}
+				size={column.size}
+			/>
+		);
+	}
+
+	const firstField = column.fields[0];
+	const rootParentField = parentField.root ?? firstField;
+	const isFieldSetOrGroup = firstField.type === 'fieldset';
+	const isFieldSet = isFieldSetOrGroup && firstField.ddmStructureId;
+	const isFieldSelected =
+		firstField.fieldName === activeField ||
+		firstField.fieldName === hoveredField;
+
+	const addr = {
+		'data-ddm-field-column': index,
+		'data-ddm-field-page': pageIndex,
+		'data-ddm-field-row': rowIndex,
+	};
+
+	return (
+		<ActionsControls
+			activePage={activePage}
+			columnRef={columnRef}
+			editable={editable}
+			expanded={isFieldSelected}
+			field={firstField}
+			ref={actionsRef}
```

Comment:
This should probably turn into an `actionsRef={actionsRef}` and we should pass the `actionsRef` to the `ref` of the `Actions` component, I think this is broken apart from making the adjustment, you can check when you do the resize.

Reply:
Yes! I'll test it too :)

Reply:
It's working ;)